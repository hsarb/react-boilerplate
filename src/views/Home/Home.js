// @flow
import React, { Component } from 'react';
import Helmet from 'react-helmet';
import newView from '../../utils/newView';
import Typography from 'material-ui/Typography';
import withStyles from 'material-ui/styles/withStyles';

const styles = {
  root: {
    alignItems: 'center',
    display: 'flex',
    justifyContent: 'center',
    flexDirection: 'column',
    minHeight: 400,
  },
};

type Props = {
  classes: Object,
  message: ?string,
};

class Home extends Component<Props> {
  static async getInitialProps() {
    const message = 'React Boilerplate';

    return { message };
  }

  render() {
    const { classes, message } = this.props;

    return (
      <div className={classes.root}>
        <Helmet title="Welcome | React-Boilerplate" />
        <Typography color="inherit" variant="display1">
          {message}
        </Typography>
        <Typography color="inherit" variant="title">
          A Minimal library react boilerplate built with performance, testing, and ease of use in
          mind.
        </Typography>
      </div>
    );
  }
}

const StyledHome = withStyles(styles)(Home);

export default newView(StyledHome);
