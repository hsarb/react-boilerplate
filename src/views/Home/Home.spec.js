import Home from './Home';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Home', () => {
  it('should render correctly', () => {
    const props = {
      message: 'Hello world',
    };

    const wrapper = shallow(<Home {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
