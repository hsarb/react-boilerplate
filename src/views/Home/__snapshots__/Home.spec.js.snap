// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<Home
  classes={
    Object {
      "root": "Home-root-1",
    }
  }
  message="Hello world"
/>
`;
